packages/uhk-web/src/app/components/agent/settings/settings.component.ts

```diff
@@ -22,11 +26,24 @@ import { ToggleAnimationEnabledAction } from '../../../store/actions/app';
 export class SettingsComponent {
     updateSettingsState$: Observable<UpdateSettingsState>;
     animationEnabled$: Observable<boolean>;
+    appTheme$: Observable<AppTheme>;
+    operatingSystem$: Observable<OperatingSystem>;
     faCog = faCog;
+    themes = [
+        { id: AppTheme.Auto, text: 'System default' },
+        { id: AppTheme.Light, text: 'Light' },
+        { id: AppTheme.Dark, text: 'Dark' }
+    ];
 
     constructor(private store: Store<AppState>) {
         this.updateSettingsState$ = store.select(appUpdateSettingsState);
         this.animationEnabled$ = store.select(getAnimationEnabled);
+        this.appTheme$ = store.select(getAppTheme);
+
+        this.operatingSystem$ = store.select(getOperatingSystem);
+        this.operatingSystem$.pipe(filter(os => os === OperatingSystem.Linux)).subscribe(() => {
```

Comment:
Selectors are the best place to calculate logics and keep the components simple as possible. Please move the `themes` calculation to a selector.
If the refactored version you also use a subscription then please unsubscribe from it on the `ngOnDestroy`

Reply:
Done.